effekt/shared/src/main/scala/effekt/core/Transformer.scala

```diff
@@ -809,7 +809,7 @@ trait TransformerOps extends ContextOps { Context: Context =>
   private[core] def bind(s: Stmt): ValueVar = {
 
     // create a fresh symbol and assign the type
-    val x = TmpValue()
+    val x = TmpValue("statementBind")
```

Comment:
Let's just call this `r` for `result` :) same for `expressionBind`. Maybe we want to prefix ALL synthesized names with something special to mark them as such. Previously `tmp` told me: "Hey this is synthesized".